WeWatch/SQLite 3/DBManager.swift

```
@@ -0,0 +1,175 @@
+//
+//  DBManager.swift
+//  WeWatch
+//
+//  Created by Anton on 07/01/2025.
+//
+//
+import Foundation
+import SQLite3
+
+internal class DBManager {
+    
+    static let shared = DBManager()
+    
+    internal enum insertError: Error {
+        
+        case movieAddError
+        case movieNotAdd
+        case dublicateError
+        case updateError
+        case notUpdate
+    }
+    
+    internal init() {
+        db = openDatabase()
+        createMovieTable()
+    }
+    
+    internal let dataPath: String = "MyDB"
+    internal var db: OpaquePointer?
+    
+    internal func openDatabase() -> OpaquePointer? {
+        let filePath = try! FileManager.default.url(
+            for: .documentDirectory,
+            in: .userDomainMask,
+            appropriateFor: nil,
+            create: false
+        ).appendingPathComponent(dataPath)
+        var db: OpaquePointer? = nil
+        if sqlite3_open(filePath.path, &db) != SQLITE_OK {
+            return nil
+        } else {
+            return db
+        }
+    }
+    
+    internal func createMovieTable() {
+        let createTableString = """
+CREATE TABLE IF NOT EXISTS Movie (
+            movieId INTEGER PRIMARY KEY,
+            title TEXT,
+            overview TEXT,
+            releaseDate TEXT,
+            rating INTEGER,
+            posterUrl TEXT
+);
+"""
+        var createTableStatement: OpaquePointer? = nil
+        if sqlite3_prepare_v2(db, createTableString, -1, &createTableStatement, nil) == SQLITE_OK {
+            if sqlite3_step(createTableStatement) == SQLITE_DONE {
+            } else {
+                print("Movie table creation failed.")
+            }
+        } else {
+            print("Movie table creation failed.")
+        }
```

Make your function throwable and throw errors instead of printing that something is fucked up.  You did it correctly in `insertMovie` function